src/inmanta/plugins.py

```diff
@@ -234,6 +228,25 @@ def resolve_type(locatable_type: LocatableString, resolver: Namespace) -> Option
     return reduce(lambda acc, transform: transform(acc), reversed(transformation_stack), resolver.get_type(locatable_type))
 
 
+class Nothing(inmanta_type.Type):
+    """
+    This custom type is used for the validation of plugins which only
+    accept null as an argument or return value.  It is meant to be used
+    as the element_type of an instance of the `NullableType` class.
+    """
+
+    def validate(self, value: object | None) -> bool:
+        return False
+
+
+# Define some types which are only used in the context of plugins.
+PLUGIN_TYPES = {
+    "any": inmanta_type.Type(),  # Any value will pass validation
+    "expression": inmanta_type.Type(),  # Any value will pass validation
+    "null": inmanta_type.NullableType(Nothing()),  # Only NoneValue will pass validation
+}
+
+
```

Comment:
@sanderr this is what I meant with "Can I define custom types for the plugin validation".  Wdyt?

Reply:
I think error messages may be messed up if you do it like this because you don't override the string representation methods (don't recall their name off the top of my head).

Reply:
The validation of arguments happens before wrapping them and the validation of the returned value happens after wrapping it, so I think it would work.  I will take the other comments though

Reply:
I'm not sure what you mean by "wrapping them". As to my other comments, I'd say have a look, but it is entirely possible that I'm wrong, it depens on how and where this is used exactly.